art/estimators/object_detection/pytorch_yolo.py

```diff
@@ -43,74 +43,78 @@ def translate_predictions_xcycwh_to_x1y1x2y2(
+    :return: Object detection labels in format x1y1x2y2 (torchvision).
     """
     import torch
 
     y_pred_x1y1x2y2 = []
-
-    for i in range(y_pred_xcycwh.shape[0]):
+    device = y_pred_xcycwh.device
+
+    for y_pred in y_pred_xcycwh:
+        boxes = torch.vstack(
+            [
+                torch.maximum((y_pred[:, 0] - y_pred[:, 2] / 2), torch.tensor(0, device=device)),
+                torch.maximum((y_pred[:, 1] - y_pred[:, 3] / 2), torch.tensor(0, device=device)),
+                torch.minimum((y_pred[:, 0] + y_pred[:, 2] / 2), torch.tensor(input_height, device=device)),
+                torch.minimum((y_pred[:, 1] + y_pred[:, 3] / 2), torch.tensor(input_width, device=device)),
+            ]
+        ).permute((1, 0))
+        labels = torch.argmax(y_pred[:, 5:], dim=1, keepdim=False)
+        scores = y_pred[:, 4]
 
         y_i = {
-            "boxes": torch.permute(
-                torch.vstack(
-                    [
-                        torch.maximum(
-                            (y_pred_xcycwh[i, :, 0] - y_pred_xcycwh[i, :, 2] / 2),
-                            torch.tensor(0).to(y_pred_xcycwh.device),
-                        ),
-                        torch.maximum(
-                            (y_pred_xcycwh[i, :, 1] - y_pred_xcycwh[i, :, 3] / 2),
-                            torch.tensor(0).to(y_pred_xcycwh.device),
-                        ),
-                        torch.minimum(
-                            (y_pred_xcycwh[i, :, 0] + y_pred_xcycwh[i, :, 2] / 2),
-                            torch.tensor(input_height).to(y_pred_xcycwh.device),
-                        ),
-                        torch.minimum(
-                            (y_pred_xcycwh[i, :, 1] + y_pred_xcycwh[i, :, 3] / 2),
-                            torch.tensor(input_width).to(y_pred_xcycwh.device),
-                        ),
-                    ]
-                ),
-                (1, 0),
-            ),
-            "labels": torch.argmax(y_pred_xcycwh[i, :, 5:], dim=1, keepdim=False),
-            "scores": y_pred_xcycwh[i, :, 4],
+            "boxes": boxes,
+            "labels": labels,
+            "scores": scores,
         }
 
         y_pred_x1y1x2y2.append(y_i)
 
     return y_pred_x1y1x2y2
 
 
-def translate_labels_art_to_yolov3(labels_art: List[Dict[str, "torch.Tensor"]]):
+def translate_labels_x1y1x2y2_to_xcycwh(
+    labels_x1y1x2y2: List[Dict[str, "torch.Tensor"]], input_height: int, input_width: int
+) -> "torch.Tensor":
     """
-    Translate labels from ART to YOLO v3 and v5.
+    Translate object detection labels from x1y1x2y2 (torchvision) to xcycwh (YOLO).
 
-    :param labels_art: Object detection labels in format ART (torchvision).
-    :return: Object detection labels in format YOLO v3 and v5.
+    :param labels_x1y1x2y2: Object detection labels in format x1y1x2y2 (torchvision).
+    :param height: Height of images in pixels.
+    :param width: Width if images in pixels.
```

Comment:
```suggestion
    :param input_height: Height of images in pixels.
    :param input_width: Width if images in pixels.
```

Reply:
I think it actually makes more sense to rename the variable than the docstring. The current names `input_height` and `input_width` are slightly redundant since it should be clear that they refer to the input image.